tests/trainer/optimization/test_parity_optimization.py

```diff
@@ -0,0 +1,286 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import os
+from unittest.mock import patch
+
+import numpy as np
+import pytest
+import torch
+import torch.nn as nn
+from torch.optim import Adam, Optimizer
+
+import pytorch_lightning as pl
+from pytorch_lightning import LightningModule, seed_everything, Trainer
+from pytorch_lightning.callbacks import ModelCheckpoint
+from pytorch_lightning.core.optimizer import LightningOptimizer
+from pytorch_lightning.utilities.exceptions import MisconfigurationException
+from pytorch_lightning.utilities.model_utils import is_overridden
+from tests.base.boring_model import BoringModel, RandomDataset, RandomDictDataset, RandomDictStringDataset
+from tests.base.models import BasicGAN
+
+"""
+TODO:
+For both automatic / manual optimization
+    - Test dp, ddp, ddp2
+    - Apex
+    - Random accumulated_grad_batches (bug)
+    - Multiple optimizers
+"""
+
+##################################################
+# TESTING AUTOMATIC OPTIMIZATION - ONE OPTIMIZER #
+##################################################
+
+################## MODULES ##################   # noqa E266
+
+
+class BaseParityAutomatiocOptimizationModel(BoringModel):
```

Comment:
Super clean I love it, just a small spelling mistake in the class names `Automatioc` -> `Automatic`

Reply:
Thanks !